src/library/scala/concurrent/Done.scala

```diff
@@ -0,0 +1,36 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2003-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package scala.concurrent
+
+/**
+ * Typically used together with `Future` to signal completion when
+ * there is no actual value completed. More clearly signals intent
+ * than `Unit` and is available both from Scala and Java (which `Unit` is not).
+ * Also helps avoiding [[http://www.scala-lang.org/files/archive/spec/2.11/06-expressions.html#value-discarding value discarding]]
+ * pitfalls such as:
+ * {{{
+ *    // Scenario: inadvertent use of `map` instead of `flatMap` for `Future` composition
+ *    //
+ *    // compiles just fine: `map[T]` is infered to be `map[Unit]`, then `Future { println("two"); Done }` which
+ *    // is of type `Future[Done]` is discarded to satisfy `map[Unit]`
+ *    def compute: Future[Unit] = Future[Done] { println("one"); Done }.map( _ => Future { println("two"); Done } )
+ *
+ *    // does not compile!
+ *    def compute: Future[Done] = Future[Done] { println("one"); Done }.map( _ => Future { println("two"); Done } )
+ * }}}
+ */
+sealed abstract class Done
```

Comment:
@backuitist Do you know of another case in the stdlib where the need for a sealed abstract class instead of a plain case object exists?

(The reason why it isn't like I suggested in Akka is because I didn't review that piece of code)